contents/tutorials/react-surveys.md

```diff
@@ -0,0 +1,504 @@
+    posthog.capture("survey dismissed", {
+      $survey_id: {surveyID} // required
+    })
+  };
+
+  const handleSubmit = (value) => {
+    setShowSurvey(false);
+    console.log("User submitted:", value);
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');  
+    posthog.capture("survey sent", {
+      $survey_id: {surveyID}, // required
+      $survey_response: {value} // required
+    })
+  };
+
+  useEffect(() => {
+    if (posthog && surveyID && showSurvey) {
+      posthog.capture("survey seen", {
+        $survey_id: {surveyID} // required
+      })
+    }
+  }, [showSurvey, surveyID, posthog])
+
+ // ... rest of your code ...
+```
+
+Altogether, your code should look like this:
+
+```js
+// src/App.js
+import { usePostHog } from 'posthog-js/react';
+import './App.css';
+import { useEffect, useState } from 'react';
+import Survey from './Survey';
+
+function App() {
+  const posthog = usePostHog()
+  const [showSurvey, setShowSurvey] = useState(true);
+  const [surveyTitle, setSurveyTitle] = useState(false);
+  const [surveyID, setSurveyID] = useState(false);
+
+  useEffect(() => {
+    posthog.getActiveMatchingSurveys((surveys) => {
+      if (surveys.length > 0) {
+        const survey = surveys[0];
+        setSurveyID(survey.id);
+        setSurveyTitle(survey.questions[0].question)
+      }
+    }); 
+  },
+  [posthog]); // posthog may be undefined until it's had a chance to initialize. Hence use it as a dependency for useEffect
+
+  useEffect(() => {
+    // Check local storage to see if the user has already seen this particular survey
+    const hasInteractedWithSurvey = localStorage.getItem(`hasInteractedWithSurvey_${surveyID}`);
+    setShowSurvey(!hasInteractedWithSurvey);
+  }, [surveyID]);
+
+  const handleDismiss = () => {
+    setShowSurvey(false);
+    console.log("Survey dismissed!");
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');
+    posthog.capture("survey dismissed", {
+      $survey_id: {surveyID} // required
+    })
+  };
+
+  const handleSubmit = (value) => {
+    setShowSurvey(false);
+    console.log("User submitted:", value);
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');  
+    posthog.capture("survey sent", {
+      $survey_id: {surveyID}, // required
+      $survey_response: {value} // required
+    })
+  };
+
+  useEffect(() => {
+    if (posthog && surveyID && showSurvey) {
+      posthog.capture("survey seen", {
+        $survey_id: {surveyID} // required
+      })
+    }
+  }, [showSurvey, surveyID, posthog])
+
+  return (
+    <div className="App">
+      <h1>This is our survey tutorial</h1>
+      {showSurvey && (
+        <Survey
+          title={surveyTitle}
+          onDismiss={handleDismiss}
+          onSubmit={handleSubmit}
+        />
+      )}
+    </div>
+  );
+}
+
+export default App;
+```
+
+That's it! Our survey is ready to go!
+
+### Option 2: Use PostHog's prebuilt survey UI
```

Comment:
It might be a lot simpler to show the prebuilt survey UI first. I gets hidden in this post otherwise.

Reply:
I thought about it and my thinking was that the custom survey aligns closer to the title "how to implement surveys in react". My concern is that putting the prebuilt survey first is a bit sneaky and pushing posthog too much